opensrp-app/src/main/java/org/smartregister/sync/helper/PlanIntentServiceHelper.java

```diff
@@ -59,6 +60,15 @@ private PlanIntentServiceHelper(PlanDefinitionRepository planRepository, Locatio
     }
 
     public void syncPlans() {
+        int batchFetchCount = batchFetchPlansFromServer();
+
+        while(batchFetchCount >= PLAN_PULL_LIMIT) {
```

Comment:
since the limit can always be changed from the server, retry sync if any record was synced